wrappers/binaries.go

```diff
@@ -21,13 +21,57 @@
 package wrappers
 
 import (
+	"io"
 	"os"
+	"regexp"
+	"strconv"
+	"strings"
 
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/snap"
 )
 
+func detectLegacy() (noCompletion bool, legacyCompletion bool) {
+	fd, err := os.Open(dirs.BashCompletionScript)
+	if err != nil {
+		return true, false
+	}
+	defer fd.Close()
+	buf := new(strings.Builder)
```

Comment:
```suggestion
	buf := &bytes.Buffer{}
```
unless you have a specific use for the builder, which I don't think there is one here

Reply:
This line has been removed.